File: typespec/common/posts.ts
import { Handle } from "./common";

export interface Post {
  id: string;
  content: string;
  tags: string[];
  author_name: string;
  author_handle: Handle;
  created_at: string;
  upvotes_count: number;
  downvotes_count: number;
  score: number;
}

File: typespec/common/posts.go
package common

import "time"

type EmployerPost struct {
	ID                 string    `json:"id"`
	Content            string    `json:"content"`
	Tags               []string  `json:"tags"`
	EmployerName       string    `json:"employer_name"`
	EmployerDomainName string    `json:"employer_domain_name"`
	CreatedAt          time.Time `json:"created_at"`
	UpdatedAt          time.Time `json:"updated_at"`
}
